### PR TITLE
Compilation of challenge 2

### PR DESCRIPTION
**Pull request by belio39**
#### What does this PR do?
Creation of Endpoints, test driven developments for endpoints of the API
#### Description of Task to be completed?
Creation of Test driven development for the endpoints and working endpoints:
<table> 
<tr>
<th>Test</th>
<th>Endpoints</th>
<th>HTTP VERBS</th>
</tr>
<tr>
<td>Post a question</td>
<td>/api/v1/questions</td>
<td>POST</td>
</tr>
<tr>
<td>View all question</td>
<td>/api/v1/questions</td>
<td>GET</td>
</tr>
<tr>
<td>Get single question</td>
<td>/api/v1/questions/<questions_id></td>
<td>GET</td>
</tr>
<tr>
<td>Post an answer to a question </td>
<td>/api/v1/questions/<question_id>/answers</td>
<td>POST</td>
</tr>
</table>

####What are the relevant Pivotal tracker stories?
[#159791228](https://www.pivotaltracker.com/story/show/159791228)
[#159873172](https://www.pivotaltracker.com/story/show/159873172)
[#159873202](https://www.pivotaltracker.com/story/show/159873202)
[#159873215](https://www.pivotaltracker.com/story/show/159873215)

### CODE REVIEW

File: UI/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "git.ignoreLimitWarning": true
```
Comment (lukyamuziB) [inline UI/.vscode/settings.json]: you don't need to commit this file

File: app/models.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+questions = []
```
Comment (lukyamuziB) [inline app/models.py:2]: opt to use a dictionary instead

**Review by lukyamuziB — CHANGES_REQUESTED** [review]
work on these changes

File: tests/test_questions.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        self.assertEqual(response.status_code, 200)
+
+    def test_no_question(self):
+        """Test to no question"""
+        response = self.client.get( 
+            '/api/v1/questions/88', data=json.dumps(self.answers), content_type='application/json')  
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.get_data())
+        self.assertEqual(data['message'], "specific question not found")
+
```
Comment (lukyamuziB) [inline tests/test_questions.py]: this applied to all your tests
`get queries` don't require a payload.

File: tests/test_questions.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        """Test to post answer"""
+        response = self.client.post(
+            '/api/v1/questions/1/answers', data=json.dumps(self.answers), content_type='application/json')
+        self.assertEqual(response.status_code, 200)
+
+    def test_no_question(self):
+        """Test to no question"""
+        response = self.client.get( 
+            '/api/v1/questions/88', data=json.dumps(self.answers), content_type='application/json')  
+        self.assertEqual(response.status_code, 200)
```
Comment (lukyamuziB) [inline tests/test_questions.py]: how come `test_no_question` returns a status code of `200`

**Review by lukyamuziB — CHANGES_REQUESTED** [review]
look into this as well

File: app/models.py (hunk 2 above)
Comment (belio39) [reply app/models.py:2]: Color

File: tests/test_questions.py (hunk 3 above)
Comment (belio39) [reply tests/test_questions.py]: Color

File: tests/test_questions.py (hunk 4 above)
Comment (belio39) [reply tests/test_questions.py]: Color please

Comment (belio39) [reply tests/test_questions.py]: `200` was the most appropriate since there was no error.

Comment (lukyamuziB) [reply tests/test_questions.py]: read about 404 and relate to what you are trying to achieve

File: tests/test_questions.py (hunk 3 above)
Comment (lukyamuziB) [reply tests/test_questions.py]: when making `GET HTTP queries`
you don't need a payload
why do your tests pass a payload and yet it's not needed

File: tests/test_questions.py (hunk 4 above)
Comment (belio39) [reply tests/test_questions.py]: I used 200 reasons being the request was processed successfully and then it gives an empty list while using 404 means the server request was not found or else the client was unable to communicate with server, in short, the request was received and successful, the 4xx codes means there was a client error whereas in this case there was none

File: tests/test_questions.py (hunk 3 above)
Comment (belio39) [reply tests/test_questions.py]: If I don't need payload on ` HTTP queries` does it mean I return null?

File: UI/.vscode/settings.json (hunk 1 above)
Comment (belio39) [reply UI/.vscode/settings.json]: >Have implemented the feedback

**Pull request merged**